src/plans/create/steps/GeneralInformationStep.tsx

```diff
@@ -0,0 +1,203 @@
+  const targetProject: string = watch(GeneralFormFieldId.TargetProject);
+  const targetProvider: V1beta1Provider = watch(GeneralFormFieldId.TargetProvider);
+  const sourceProvider: V1beta1Provider = watch(GeneralFormFieldId.SourceProvider);
+  const [activeNamespace] = useActiveNamespace();
+  const defaultNamespace = getDefaultNamespace();
+  const initialProjectName =
+    planProject || (activeNamespace === Namespace.AllProjects ? defaultNamespace : activeNamespace);
+  const projectOptions = useProjectNameSelectOptions();
+  const [targetProviderProjects, targetProviderProjectsLoading] =
+    useProviderNamespaces(targetProvider);
+
+  return (
+    <WizardStepContainer title={t('General')}>
+      <Form>
+        <FormSection title={t('Plan information')} titleElement="h3">
+          <p>{t('Name your plan and choose the project you would like it to be created in.')}</p>
+
+          <ControlledFormGroup
+            isRequired
+            fieldId={GeneralFormFieldId.PlanName}
+            label={generalFormFieldLabels[GeneralFormFieldId.PlanName]}
+            controller={{
+              rules: { required: t('Plan name is required.') },
+              render: ({ field }) => (
+                <TextInput
+                  {...field}
+                  validated={
+                    errors[GeneralFormFieldId.PlanName]
+                      ? ValidatedOptions.error
+                      : ValidatedOptions.default
+                  }
+                />
+              ),
+            }}
+          />
+
+          <ControlledFormGroup
+            isRequired
+            fieldId={GeneralFormFieldId.PlanProject}
+            label={generalFormFieldLabels[GeneralFormFieldId.PlanProject]}
+            controller={{
+              ...(projectOptions?.find((option) => option.value === initialProjectName) && {
+                defaultValue: initialProjectName,
+              }),
+              rules: { required: t('Plan project is required.') },
+              render: ({ field }) => (
+                <div ref={field.ref}>
+                  <TypeaheadSelect
+                    placeholder={t('Select plan project')}
+                    id={GeneralFormFieldId.PlanProject}
+                    selectOptions={projectOptions}
+                    selected={field.value}
+                    onSelect={(_, value) => {
+                      field.onChange(value);
+                      setProviderNamespace(value.toString());
+
+                      if (sourceProvider) {
+                        setValue(GeneralFormFieldId.SourceProvider, '', { shouldValidate: true });
+                      }
+                      if (targetProvider) {
+                        setValue(GeneralFormFieldId.TargetProvider, '', { shouldValidate: true });
+                      }
+                      if (targetProject) {
+                        setValue(GeneralFormFieldId.TargetProject, '', { shouldValidate: true });
+                      }
```

Comment:
please explain

Reply:
reseting other values each time NS for plan is changed?

Reply:
Exactly, yes, but we only want to reset them if those values already exist, hence the conditions around each. And we are also validating those fields after the reset so that their respective error messages are shown indicating to the user they need to re-enter them.

Reply:
resolving this conversation since it's a duplicate now of https://github.com/kubev2v/forklift-console-plugin/pull/1547#pullrequestreview-2731219957.